File: app/src/main/java/ru/merkulyevsasha/news/data/db/mappers/ArticleEntityMapper.java
package ru.merkulyevsasha.news.data.db.mappers;

import ru.merkulyevsasha.news.data.Mapper;
import ru.merkulyevsasha.news.data.db.entities.ArticleEntity;
import ru.merkulyevsasha.news.pojos.Article;

public class ArticleEntityMapper implements Mapper<ArticleEntity, Article> {
    @Override
    public Article map(ArticleEntity item) {
        return new Article(item.getId(), item.getSourceNavId(), item.getTitle(), item.getLink()
                , item.getDescription(), item.getPubDate(), item.getCategory(), item.getSearch(), item.getPictureUrl());
    }
}

File: app/src/main/java/ru/merkulyevsasha/news/data/db/mappers/ArticleEntityMapper.kt
package ru.merkulyevsasha.news.data.db.mappers

import ru.merkulyevsasha.news.data.Mapper
import ru.merkulyevsasha.news.data.db.entities.ArticleEntity
import ru.merkulyevsasha.news.models.Article

class ArticleEntityMapper : Mapper<ArticleEntity, Article> {
    override fun map(item: ArticleEntity): Article {
        return Article(
                item.id,
                item.sourceNavId,
                item.title,
                item.link,
                item.description,
                item.pubDate,
                item.category,
                item.search,
                item.pictureUrl
        )
    }
}
